core/src/main/resources/reference.conf

```diff
@@ -420,6 +420,18 @@ datastax-java-driver {
     #   change.
     # Overridable in a profile: no
     warn-on-init-error = true
+
+    # The timeout to use when establishing driver connections.
+    #
+    # This timeout is for controlling how long the driver will wait for the underlying channel
+    # to actually connect to the server. This is not the time limit for completing protocol
+    # negotiations, only the time limit for establishing a channel connection.
+    #
+    # Required: yes
+    # Modifiable at runtime: yes, the new value will be used for connections created after the
+    #   change.
+    # Overridable in a profile: no
+    connect-timeout = 5 seconds
```

Comment:
Nit: could you move it at the beginning of the section, either before or after the other timeouts? It feels slightly more convenient to have all the connection timeouts in the same place.